src/app/features/parent/buy-tickets/buy-tickets.component.ts

```diff
@@ -50,21 +50,26 @@ export class BuyTicketsComponent implements OnInit {
             this.eventService.getSections(this.eventId).subscribe(data => {
                 // assuming this data is ordered with section locations ascending
                 this.imageURLs = data.map((section: Section) => section.imageURL);
-                this.selectSection('top');
-
+                this.selectSection(this.pathToLocation.get("top")!);
             });
 
         });
 
     }
 
+    updateVRImage(elementId: number) {
+      if (elementId != undefined) {
+        this.imageSelector = elementId;
+      }
+    }
+
     buyTicket() {
         this.userService.buyTicket(this.selectedTicket.id).subscribe(() => {
             this.router.navigate(['/my-tickets'])
         });
     }
 
-    selectSection(elementId: string) {
+    selectSection(elementId: number) {
```

Comment:
You've change all calls to this method to now have to use `pathToLocation.get('')`. if every call to a method requires you to call another method in the the call something isn't right (also generally you want function calls in html to as simple as possible, so no nesting calls):
`this.selectSection(this.pathToLocation.get("top")!);`
vs
`this.selectSection("top");`
Just leave this as it was, because now you're essentially using the elementId to get some other value to then just get the elementId again from a list, why not just give it the elementId in the first place?
```
selectSection(elementId: string) {
        // Deselect other sections
        this.pathIds.forEach(id => {
            let element = document.getElementById(id);
            if (element) {
                element.classList.remove('selected');
            }
        })
        // Select this section
        let element = document.getElementById(elementId);
        console.log(element)
        if (element) {
            element.classList.add('selected');
        }

        // Update VR Image
        const index = this.pathToLocation.get(elementId);
        if (index != undefined) {
            this.imageSelector = index;
        }
}
```